sunpy/data/manager/manager.py

```diff
@@ -0,0 +1,135 @@
+import pathlib
+import functools
+from contextlib import contextmanager
+
+from sunpy.util.util import hash_file
+
+
+class DataManager:
+    """
+    DataManager
+    """
+
+    def __init__(self, cache):
+        # TODO: Folder prefix should be init argument
+        self._cache = cache
+
+        self._file_cache = {}
+
+        self._skip_hash_check = False
+        self._skip_file = {}  # Dict[str, str]
+
+    def require(self, name, urls, sha_hash):
+        """decorator for doing stuff
+
+        Parameters
+        ----------
+        name: str
+        The name to reference the file with
+        urls: list
+        List of urls to download the file from
+        sha_hash: str
+        Hash of file
+        """
+        def decorator(func):
+            @functools.wraps(func)
+            def wrapper(*args, **kwargs):
+                # TODO: Refractor into function(s)
+                replace = self._skip_file.get(name, None)
+                if replace:
+                    if replace.startswith('file://'):
+                        file_path = replace[len('file://'):]
+                    else:
+                        file_path = self._cache.download([replace], redownload=True)
+                elif self._skip_hash_check:
+                    # XXX: Should this redownload every time
```

Comment:
Would redownloading only if the file is missing work?

Reply:
Yeah. With a cache expiry time. That's what was in my mind.

Ping: @Cadair

Reply:
Yeah I think that's fair, if we aren't validating the hash then we really only have the filename to go on.